core/types/calls.cc

```diff
@@ -696,7 +696,20 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, DispatchArgs args, core
     // the send, assume that the last argument is an implicit keyword args hash.
     bool implicitKwsplat = false;
     if (ait != aPosEnd && hasKwargs && args.args.size() == args.numPosArgs) {
-        // NOTE: this would be a good place for an autocorrect to using `**kwhash`
+        auto splatLoc = core::Loc(args.locs.file, args.locs.args[args.args.size() - 1]);
+
+        // If --error-kwarg-hash-without-splat is set, we will treat "**-less" keyword hash argument as an error.
+        if (gs.errorKwargHashWithoutSplat) {
+            if (auto e = gs.beginError(splatLoc, errors::Infer::KeywordArgHashWithoutSplat)) {
+                e.setHeader("Keyword argument hash without `{}` is deprecated", "**");
+                e.addErrorLine(splatLoc, "This produces a runtime warning in Ruby 2.7, "
+                                         "and will be an error in Ruby 3.0");
+                e.addErrorLine(splatLoc, "Note: disabling `{}` will suppress this error",
+                               "--error-kwarg-hash-without-splat");
```

Comment:
How about removing this line? Since it's off by default, It would be something that the user would know they had enabled already.

Reply:
Sure, that sounds good.